drivers/mdio/mdio_shell.c

```diff
@@ -168,26 +154,25 @@ static int cmd_mdio_read(const struct shell *sh, size_t argc, char **argv)
 	return 0;
 }
 
-/* mdio write_c45 <port_addr> <dev_addr> <reg_addr> <value> */
+/* mdio write_c45 <device> <port_addr> <dev_addr> <reg_addr> <value> */
 static int cmd_mdio_write_45(const struct shell *sh, size_t argc, char **argv)
 {
 	const struct device *dev;
 	uint16_t data;
 	uint16_t reg_addr;
 	uint8_t dev_addr;
 	uint8_t port_addr;
+	int ret;
 
-	dev = DEVICE_DT_GET(MDIO_NODE_ID);
-	if (!device_is_ready(dev)) {
-		shell_error(sh, "MDIO: Device driver %s is not ready.", dev->name);
-
-		return -ENODEV;
+	ret = parse_device_arg(sh, argv, &dev);
+	if (ret < 0) {
+		return ret;
 	}
 
-	port_addr = strtol(argv[1], NULL, 16);
-	dev_addr = strtol(argv[2], NULL, 16);
-	reg_addr = strtol(argv[3], NULL, 16);
-	data = strtol(argv[4], NULL, 16);
+	port_addr = strtol(argv[2], NULL, 16);
+	dev_addr = strtol(argv[3], NULL, 16);
+	reg_addr = strtol(argv[4], NULL, 16);
+	data = strtol(argv[5], NULL, 16);
```

Comment:
It could be worth checking whether the conversion was successful, though I understand that's beyond the scope of this change.